randovania/gui/dialog/connections_editor.py

```diff
@@ -302,9 +302,11 @@ def current_requirement(self) -> RequirementArrayBase | None:
         if comment == "":
             comment = None
 
-        nested = [editor.current_requirement for editor in self._editors]
-        if any(it is None for it in nested):
-            return None
+        nested: list[Requirement] = []
+        for it in nested:
+            if it is None:
+                return None
+            nested.append(it)
```

Comment:
With every OS failing on the tests, I guess they need to be fixed.
Is it intentional that `nested` is now always an empty array? That doesn't look right to me - without looking into the rest of the context - and can't see how this loop is every executed.

Reply:
WTF DID I WRITE HERE